shared/bar/src/commonMain/kotlin/com/tappchart/bar/BarChart.kt

```diff
@@ -1,27 +1,224 @@
+@file:OptIn(ExperimentalTextApi::class)
+
 package com.tappchart.bar
 
+import androidx.compose.foundation.Canvas
 import androidx.compose.runtime.Composable
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.geometry.Size
 import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.drawscope.DrawScope
+import androidx.compose.ui.text.ExperimentalTextApi
+import androidx.compose.ui.text.TextLayoutResult
+import androidx.compose.ui.text.TextMeasurer
 import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.drawText
+import androidx.compose.ui.text.rememberTextMeasurer
 import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
 import com.tappchart.bar.model.Bar
 import com.tapptitude.tappchart.model.AxisInterval
 import com.tapptitude.tappchart.model.ValueLabel
 import com.tapptitude.tappchart.ui.AxisStyle
 import com.tapptitude.tappchart.ui.BackgroundStyle
-
+import com.tapptitude.tappchart.ui.drawBackgroundGradient
 
 @Composable
 fun BarChart(
     modifier: Modifier = Modifier,
     yAxisInterval: AxisInterval,
     data: List<Bar>,
-    verticalLabel: List<ValueLabel>,
+    verticalLabels: List<ValueLabel>,
     labelStyle: TextStyle,
     spacingBetweenBars: Dp,
     axisStyle: AxisStyle,
-    background: List<BackgroundStyle> = listOf(),
     barColor: List<Color>,
+    background: List<BackgroundStyle> = listOf(BackgroundStyle.HorizontalGrid),
+    yAxisPaddingToLine: Dp = 8.dp,
+    xAxisPaddingToBars: Dp = 2.dp,
+    firstBarStartPadding: Dp = 0.dp,
+    lastBarEndPadding: Dp = 0.dp,
+    showTopLabelMisaligned: Boolean = false
+) {
+
+    val textMeasurer = rememberTextMeasurer()
+    Canvas(modifier = modifier) {
+        val horizontalLabelHeight = textMeasurer.measure(data[0].key.label).size.height
```

Comment:
We could use `.first()` instead of `data[0]`. Also, we might want to make a check to make sure the data set is not empty. We can either not draw anything if it is empty or we could leverage `require` and throw a runtime error